srtcore/channel.cpp

```diff
@@ -786,34 +773,62 @@ int srt::CChannel::sendto(const sockaddr_any& addr, CPacket& packet, const socka
 
     const int res = (int)::sendmsg(m_iSocket, &mh, 0);
 #else
-    DWORD size     = (DWORD)(CPacket::HDR_SIZE + packet.getLength());
+    DWORD size     = (DWORD)(packet.m_PacketVector[0].size() + packet.m_PacketVector[1].size());
     int   addrsize = addr.size();
+    class WSAEventRef
+    {
+    public:
+        WSAEventRef()
+            : e(::WSACreateEvent())
+        {
+        }
+        ~WSAEventRef()
+        {
+            ::WSACloseEvent(e);
+            e = NULL;
+        }
+        void reset()
+        {
+            ::WSAResetEvent(e);
+        }
+        WSAEVENT Handle()
+        {
+            return e;
+        }
 
-    int res = ::WSASendTo(m_iSocket, (LPWSABUF)packet.m_PacketVector, 2, &size, 0, addr.get(), addrsize, &m_SendOverlapped, NULL);
-
+    private:
+        WSAEVENT e;
+    };
+#if HAVE_CXX11
+    thread_local WSAEventRef lEvent;
+#else
+    WSAEventRef lEvent;
```

Comment:
Why do we keep the wrong solution in case of non-C++11 build?

If we state that the Windows version can only be compiled with C++11, and this feature is exclusively for Windows, then we don't need any alternative here.

Reply:
It is not completely wrong, it just creates and deletes a local object on each call for non-C++11 Windows builds.
This would also be the case for the minGW build.
Using `pthread_once` would be of course better.

Reply:
Ok, the situation is this: we can just as well disable "old" builds on MICROSOFT (that is, Windows with MSVC), but there is still possible MinGW with C++03, the possibility to compile successfully with C++11 (and C++11 sync) exists for MinGW, but this doesn't always work.

In short: we can disable C++03 builds on Windows, but not on MinGW. And if with MinGW, then we need a C++03 alternative for `thread_local` with the use of `pthread_once`.

Reply:
Ok, the situation is this: we can just as well disable "old" builds on MICROSOFT (that is, Windows with MSVC), but there is still possible MinGW with C++03, the possibility to compile successfully with C++11 (and C++11 sync) exists for MinGW, but this doesn't always work.

Still, I don't get why you apply `thread_local` onto a local variable.

Reply:
Likely ` thread_local static` was meant.